### PR TITLE
JSON extensions

### PR DESCRIPTION
**Pull request by RagingLink**
Allow users to access values and keys of objects programmatically with `jsonkeys` and `jsonvalues`. Implementation-wise, this is no different from `{json}` or `{jsonget}` (code is practically indifferent), but it returns (obviously) the values or keys of the object instead.
Also implemented `jsonsort` which sorts an array of objects based on properties in the objects.
**Added:**
- `jsonvalues` and `jsonkeys` [d7ae7f8](https://github.com/blargbot/blargbot/commit/d7ae7f82c2895d776a59025497e791a15872bd88)
- `jsonsort` [ab5c5e0](https://github.com/blargbot/blargbot/pull/222/commits/ab5c5e03288a80fe9e31ca9cbfd47b3ff578e013)

### CODE REVIEW

File: src/tags/jsonkeys.js
Comment (HunteRoi) [inline src/tags/jsonkeys.js:17]: You should review this description because a JSON object is by definition the notation of a Javascript object as a string. This will lead to confusion for advanced developers or people used to bbtag being a language without "real types"

File: src/tags/jsonsort.js
Comment (HunteRoi) [inline src/tags/jsonsort.js:23]: why do you jsonstringify the response of jsonsort in the example?

File: src/tags/jsonsort.js
Comment (HunteRoi) [inline src/tags/jsonsort.js:39]: your callback function does not return an array of Promises so I don't understand why you have a `Promise.all` here. Can you explain this choice? @RagingLink 

cc @Ratismal

File: src/tags/jsonkeys.js
Comment (Ratismal) [inline src/tags/jsonkeys.js:16]:
```suggestion
        .withDesc('Retrieves all keys from provided the JSON object. ' + 
```

File: src/tags/jsonkeys.js
Comment (Ratismal) [reply src/tags/jsonkeys.js:17]: I think it's fine, in this context it needs to be an object in JSON notation (ex. quotes keys, double quotes, etc.)

File: src/tags/jsonvalues.js
Comment (Ratismal) [inline src/tags/jsonvalues.js:16]:
```suggestion
        .withDesc('Retrieves all values from the provided object. ' +
```

File: src/tags/jsonsort.js
Comment (Ratismal) [reply src/tags/jsonsort.js:39]: Yeah, this definitely doesn't need to be run with Promise.all

File: src/tags/jsonsort.js
Comment (Ratismal) [inline src/tags/jsonsort.js:16]:
```suggestion
        .withDesc('Sorts an array of objects based on the provided `path`.\n' +
```

File: src/tags/jsonsort.js
Comment (Ratismal) [inline src/tags/jsonsort.js:17]:
```suggestion
            '`path` is a dot-noted series of properties.\n' +
```

**Review by Ratismal — APPROVED** [review]
A few nitpicks in documentation that I'll commit, will also clean up some other stuff (ex. promise logic) after merging.

**Pull request merged**
